src/main/java/org/mitre/synthea/export/BB2RIFExporter.java

```diff
@@ -2569,46 +2607,88 @@ private long exportHome(Person person, long startTime, long stopTime) throws IOE
+          String.format("%.2f", servicePeriod.getTotalCost().getCoveredCost()));
       fieldValues.put(HHA.CLM_TOT_CHRG_AMT,
-          String.format("%.2f", encounter.claim.getTotalClaimCost()));
-      fieldValues.put(HHA.CLM_HHA_TOT_VISIT_CNT, "" + homeVisits);
-      int days = (int) ((encounter.stop - encounter.start) / (1000 * 60 * 60 * 24));
+          String.format("%.2f", servicePeriod.getTotalCost().getTotalClaimCost()));
+      fieldValues.put(HHA.CLM_HHA_TOT_VISIT_CNT, "" + servicePeriod.getEncounters().size());
+      int days = (int) ((servicePeriod.getStop() - servicePeriod.getStart())
+              / (1000 * 60 * 60 * 24));
       if (days <= 0) {
         days = 1;
       }
       fieldValues.put(HHA.REV_CNTR_UNIT_CNT, "" + days);
       fieldValues.put(HHA.REV_CNTR_RATE_AMT,
-          String.format("%.2f", encounter.claim.getTotalClaimCost()
+          String.format("%.2f", servicePeriod.getTotalCost().getTotalClaimCost()
                   .divide(BigDecimal.valueOf(days), RoundingMode.HALF_EVEN)
                   .setScale(2, RoundingMode.HALF_EVEN)));
       fieldValues.put(HHA.REV_CNTR_PMT_AMT_AMT,
-          String.format("%.2f", encounter.claim.getCoveredCost()));
+          String.format("%.2f", servicePeriod.getTotalCost().getCoveredCost()));
       fieldValues.put(HHA.REV_CNTR_TOT_CHRG_AMT,
-          String.format("%.2f", encounter.claim.getTotalClaimCost()));
+          String.format("%.2f", servicePeriod.getTotalCost().getTotalClaimCost()));
       fieldValues.put(HHA.REV_CNTR_NCVRD_CHRG_AMT,
-          String.format("%.2f", encounter.claim.getPatientCost()));
+          String.format("%.2f", servicePeriod.getTotalCost().getPatientCost()));
 
+      // random from fields TSV, may be overriden below
+      String revCenter = fieldValues.get(HHA.REV_CNTR);
+
+      final String HHA_TOTAL_CHARGE_REV_CNTR = "0001"; // Total charge
+      final String HHA_MEDICATION_REV_CNTR = "0270"; // General medical/surgical supplies
+      final String HHA_MEDICATION_CODE = "T1502"; // Administration of medication
+      ConsolidatedClaimLines consolidatedClaimLines = new ConsolidatedClaimLines();
+      for (HealthRecord.Encounter encounter : servicePeriod.getEncounters()) {
+        for (ClaimEntry lineItem : encounter.claim.items) {
+          String hcpcsCode = null;
+          if (lineItem.entry instanceof HealthRecord.Procedure) {
+            for (HealthRecord.Code code : lineItem.entry.codes) {
+              if (hcpcsCodeMapper.canMap(code.code)) {
+                hcpcsCode = hcpcsCodeMapper.map(code.code, person, true);
+                if (hhaRevCntrMapper.canMap(code.code)) {
+                  revCenter = hhaRevCntrMapper.map(code.code, person);
+                }
+                break; // take the first mappable code for each procedure
+              }
+            }
+            if (hcpcsCode == null) {
+              revCenter = HHA_TOTAL_CHARGE_REV_CNTR;
+            }
```

Comment:
I might be reading this wrong, but it looks like multiple line items per claim that do not have a HCPCS mapping could have the total charge rev center, but don't you only want that once per claim?

Reply:
Right, all of the events with a null HCPCS code get rolled up into the 0001 rev center. Note that REV_CNTR_UNIT_CNT indicates how many.